ARCHITECTURE.md

```diff
@@ -158,10 +173,115 @@ This button with type submit inside a form will make it submit on ENTER key pres
 Form components located at `src/forms` are already setup to use with `react-hook-form` together with `chakra` by passing only the form's `control`
 
 
-### State Management ( Jotai )
+## State Management ( Jotai )
 
 We've recently included [Jotai](https://jotai.org/docs/core/atom) as one of the modular tools for handling state managment. Jotai creates modular atoms and derived atoms that are accessible throughout the app through the atom. 
 
 Our strategy is to migrate a large part of context overheads to Jotai for more effective state management. This move aims to boost the speed of state updates and renders.
 
 In any situation where a new state is needed across multiple scopes, we recommend using Jotai `Atoms`. Jotai’s primitive and declarative API makes handling complex state needs more manageable. It facilitates splitting the global state into a collection of distributed but well synchronized atoms, bringing more efficient and maintainable state management to the project.
+
+
+## Asset Handling
+
+
+Images uploaded by creators in Geyser are automatically resized into three distinct sizes: small, medium, and large. These images share the same URL, differentiated by postfixes such as `image_small.webp`, `image_medium.webp` and `image_large.webp`.
+
+For instance, if a user uploads an image named 'profile.png', it's uploaded, and the returned value is the small image in the format `https://geyserbucket...../profile_image_small.webp`. This process is automatic, allowing us to add the image directly to the database without manual manipulation.
+
+However, there are situations where a larger image size is required, like in a project thumbnail popup. In such cases, the small image URL needs to be converted to a large image URL, i.e., `https://geyserbucket...../profile_image_large.webp`.
+
+To facilitate this, we've provided functions in the utils folder: `toLargeImageUrl`, `toMediumImageUrl`, and `toSmallImageUrl`. These functions can be used to display images of different sizes as necessary. Even if the incoming image size is unknown, these functions act as filters, ensuring the correct image size renders.
+
+
+
+## Architecture Goal 
+
+The geyser-app is currently being restructured to improve maintainability, readability, and performance.
+Guided by Domain Driven Design (DDD) principles and the Horizontal Split principle, we'll reorganize all pages and components into Modules.
```

Comment:
It'd be great to add links to extra resources about Horizontal Architecture/DDD. For DDD, we have a Notion page you can link to: https://www.notion.so/geyser/Domain-Driven-Design-DDD-8c080fee21c644569e99ce1d0d9e8e99